src/RecognitionClient.cpp

```diff
@@ -135,19 +142,22 @@ RecognitionClient::buildRecognitionConfig(const Configuration& configuration) {
     return recognitionConfig;
 }
 
-speechcenter::recognizer::v1::RecognitionStreamingRequest
-RecognitionClient::buildAudioRequest(const Configuration& configuration) {
+std::vector<speechcenter::recognizer::v1::RecognitionStreamingRequest>
+RecognitionClient::buildAudioRequests(const Configuration& configuration) {
     INFO("Building audio request...");
-
-    Audio audio(configuration.getAudioPath());
-
+    AudioFile audioFile(configuration.getAudioPath()) ;
+    auto const &audio = audioFile.getAudio();
     int64_t lengthInBytes = audio.getLengthInBytes();
-
     INFO("Audio bytes: " + std::to_string(lengthInBytes));
-    speechcenter::recognizer::v1::RecognitionStreamingRequest audioRequest;
-    audioRequest.set_audio(static_cast<void*>(audio.getData()), lengthInBytes);
 
-    return audioRequest;
+    std::vector<speechcenter::recognizer::v1::RecognitionStreamingRequest> requests;
+    auto chunks = audio.getAudioChunks<2000>();
+    for (const auto chunk : chunks) {
```

Comment:
i think that:
`for (const auto& chunk : chunks) {`
a reference  for each chunk will be better

Reply:
Agree. changed.